src/katsdpnetboxutilities/system/device-md.py

```diff
@@ -0,0 +1,469 @@
+                    data["partitions"][dev["name"]] = dev
+
+        return data
+
+    def _remote_get(self, path, filename, is_json=True):
+        if path:
+            _url = "{}/{}/{}".format(
+                self.url.strip("/"), path.strip("/"), filename.strip("/")
+            )
+        else:
+            _url = "{}/{}".format(self.url.strip("/"), filename.strip("/"))
+        req = requests.get(_url)
+        if req.status_code == requests.codes.ok:
+            if is_json:
+                return req.json()
+            else:
+                return req.body
+        else:
+            logging.warning("Could not load - %s", _url)
+
+    def get_file_from_remote(self, filename, is_json=True):
+        # TODO: fix the retry
+        if self.url is None:
+            logging.warning("No remote server defined, not fetching from remote.")
+            return {}
+        data = self._remote_get(self.device_name, filename, is_json)
+        if data is None:
+            data = self._remote_get(
+                "servers/{}".format(self.device_name), filename, is_json
+            )
+        if data is None:
+            data = self._remote_get(None, filename, is_json)
+        if not data:
+            logging.warning("No %s found on %s", filename, self.url)
+            data = {}
+        return data
+
+    def get_lshw(self) -> dict:
+        lshw = self._cache.get("lshw")
+        if lshw is None:
+            lshw = self.get_file_from_remote("lshw.json")
+            self._cache["lshw"] = lshw
+        return lshw or {}
+
+    def get_lsblk(self) -> dict:
+        lsblk = self._cache.get("lsblk")
+        if lsblk is None:
+            lsblk = self.get_file_from_remote("lsblk.json")
+            self._cache["lsblk"] = lsblk
+        return lsblk or {}
+
+    def lshw_core(self):
+        """Helper method to return only the lshw core"""
+        lshw = self.get_lshw()
+        for child in lshw.get("children", []):
+            if child.get("id") == "core":
+                return child
+        return {}
+
+    def lsblk_devices(self):
+        lsblk = self.get_lsblk()
+        data = {}
+        for blkdev in lsblk.get("blockdevices", []):
+            pprint(blkdev)
+            if blkdev.get("type") not in ["loop"]:
+                data[blkdev["name"]] = blkdev
+        return data
+
+
+class Page:
+    def __init__(self):
+        self._lines = []
+
+    def heading(self, heading, level):
+        self._lines.append("#" * level + " " + heading)
+        self._lines.append(None)
+
+    def text(self, text):
+        self._lines.append(text)
+
+    def ll_table(self, rows):
+        """List-List no header table"""
+        # Not a table just text now.
+        self._lines.append(None)
+        self._lines.append("**Specifications**\n")
+        for row in rows:
+            self._lines.append(f"\t - {row[0]} : **{row[1]}**\n".strip())
+        self._lines.append(None)
+
+    def write(self, filename):
+        if filename:
```

Comment:
I think we should write out the yaml page header. See the Pandoc docs.
This will be a dictionary in this class that will e written out first starting with '---' and ending with '...'.
Seems there are a lot of settings that is controlled this way and it is  more contained than the header.tex file. The question then is, can we replace header.tex with this/ Can we have both?

Reply:
I notice this, and added a crude header section to get the pages to be A4. Probably other way, can you change the page style from letter to A4 please.